Add Fathom Analytics Stats extension

yannglt opened the pull request:
## Description

A Raycast extension to browse your website analytics coming from Fathom.

## Screencast
(Don't wanted to share my website analytics so here are screenshots of the extension running with pixelated areas.)
![fathom-analytics-stats-1](https://github.com/raycast/extensions/assets/16894359/2505bdd3-709c-4994-bcc3-e10edf0a14a6)
![fathom-analytics-stats-2](https://github.com/raycast/extensions/assets/16894359/abb6fdc9-6605-49b9-8c61-ef76c3abed8c)
![fathom-analytics-stats-3](https://github.com/raycast/extensions/assets/16894359/1a050f3a-1e28-4c32-bb8d-f924905ee47f)

## Checklist
- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

yannglt commented:
Added more detailed instructions to get an API token from Fathom (cc @peduarte, was not really clear indeed)

pernielsentikaer commented:
Hi 👋

Thanks for your contribution 🔥 

Wouldn't it make sense to add to the current [Fathom Analytics](https://www.raycast.com/danielferguson/fathom-analytics) extension we already has in the Store by @DanielFerguson 🙂

pernielsentikaer commented:
After checking them both out, it actually makes sense to keep it as two separate extensions. We should keep this as it is.

@DanielFerguson, I DM'ed you in Slack and X. Do you have a minute? :)

yannglt commented:
Hey Per ✌️

Thank you for your time and review of this new extension.

Let me know if you need anything from my side before merging. Happy to help if needed!

pernielsentikaer commented on extensions/fathom-analytics-stats/src/open-site-settings.tsx:
```diff
@@ -0,0 +1,10 @@
+import { getPreferenceValues, open } from "@raycast/api";
+
+interface Preferences {
+  apiToken: string;
+  siteId: string;
+}
```
```suggestion
```
You can remove this in all files since it's already automatic generated by Raycast's `raycast-env.d.ts`

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- The extension crashes when you enters wrong credemtials, could you handle this gracefully?

![fathom-analytics-stats 2024-01-25 at 08 57 13](https://github.com/raycast/extensions/assets/38033107/2fde3ed6-8ecb-4d44-9502-cd8e9da5c7ef)

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

yannglt commented:
Hey Per 👋

I added error handling for the extension with a `Toast` and a `Detail` view to provide more details.  
Also added some improvements (types, time period order, setup guide).

Let me know if you need anything else.

peduarte commented:
🔥 🔥 🔥 🔥 🔥

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥
